### DISCUSSION TITLE
Backup to URL and Directory

### DISCUSSION THREAD

**Discussion by johnfinlay** [olahallengren/sql-server-maintenance-solution#502] [Questions & Answers] [answered]
Can @URL and @Directory be used at the same time for Backup?

**Comment by Directory** [comment]
can you learn to use codeblocks and not ping random people

**Reply by johnfinlay** [reply]
Can you learn to use a real name and not some random word that’s used commonly in IT?

**Reply by olahallengren** [reply]
@Directory, is there anything that I can do to prevent you from getting the notifications? Is there any configuration in GitHub?

```@Directory``` is one of the most commonly used parameters in this project and I suspect that you will continue to get notifications as users are not always using code blocks.

**Reply by Directory** [reply]
@johnfinley no disrespect john but you seem to be a normal middle aged dude using GitHub for career purposes and are not too adjusted to internet culture. One day i strive to be like you. I am definitely not ever using my real name here. But for me this is the choice of handle i use. 

As for what can be done everybody could use codeblocks when they copy paste their issues. It's not only proper but more readable. Im not the only one getting pinged. Tons of other vanity account names are too. I myself don't really care anymore. I don't use GitHub in a proffesional capacity and just Ignore the daily spam emails. Nobody needs to put up with me.

**Reply by johnfinlay** [reply]
Fair enough, I didn’t consider codeblocks since I was just referring to parameters in my question. Lesson learned on my part.

**Comment by olahallengren** [comment, marked as answer]
It is not possible to combine backup to disk and URL.

SQL Server only supports backup striping and backup mirroring to the same type of storage.

**Reply by mkless** [reply]
Dear Ola,

we love to use your SQL Server Maintenance Solution for backup, integrity checks and statistics maintenance!

Actually we are going to implement an Hybrid Backup solution for our SQL Server landscape, that is we want to backup to azure storage for long-term keeping in addition to mirroring backups to local on-prem storage for short-term access. The idea to use the mirroring option of the backup command, to backup to disk and mirror to Azure-URL.

The generic backup command works when using the parameters format and blocksize:

BACKUP DATABASE [AdventureWorks2012] 
TO  Disk = N''D:\Backup\AdventureWorks.bak'
mirror to URL= 'https://anyazurestorageaccount.blob.core.windows.net/backup/AdventureWorks.bak'
with format, blocksize=65536

When we try the same using your DatabaseBackup stored procedure, the command fails due to different checks of URLs and Directory parameter of the stored procedure.

for example:
Msg 50000, Level 16, State 1, Procedure AdminDB.dbo.DatabaseBackup, Line 2276 [Batch Start Line 0]
The value for the parameter @Directory is not supported.
It is not possible to mix URL and MIRRORURL  with DIRECTORY and MIRRORDIRECTORY parameter and vice versa.

Do you have any suggestion for us?

Thanks in Advance

Michael

**Reply by johnfinlay** [reply]
I would suggest backup to disk and verify, then use something like Powershell to copy the backup file to Azure
